Upgrade EUI to v54.0.0

cee-chen opened the pull request:
## Summary

`eui@53.0.1`  ⏩  `eui@54.0.0`

## [`54.0.0`](https://github.com/elastic/eui/tree/v54.0.0)

- `EuiDataGrid` now allows limiting the number of visible cell actions with a new `columns.visibleCellActions` prop (defaults to 2). All additional actions will be shown in the cell expansion popover. ([#5675](https://github.com/elastic/eui/pull/5675))
- Added a new `gridStyle.rowClasses` API to `EuiDataGrid`, which allows adding custom classes/styling to specific row indices, primarily for the purpose of highlighting rows ([#5732](https://github.com/elastic/eui/pull/5732))
- Added `alert` icon indicator and `aria-invalid` when the following form controls are `isInvalid`: `EuiFieldNumber`, `EuiFieldPassword`, `EuiFieldText`, `EuiSelect`, `EuiSuperSelect`, `EuiFieldSearch`, `EuiColorPicker` ([#5738](https://github.com/elastic/eui/pull/5738))
- Added `isInvalid` prop to `EuiFormControlLayout` to render the `alert` icon ([#5738](https://github.com/elastic/eui/pull/5738))
- Added `isDropdown` prop to `EuiFormControlLayout` to create and control an `arrowDown` icon ([#5738](https://github.com/elastic/eui/pull/5738))
- Added `color` as to `EuiFormControlLayout`'s `icon` object ([#5738](https://github.com/elastic/eui/pull/5738))

**Bug fixes**

- Fixed `EuiSuperSelect` border-radius with `append` or `prepend` ([#5738](https://github.com/elastic/eui/pull/5738))
- Fixed `EuiSuperSelect` not respecting `readOnly` ([#5738](https://github.com/elastic/eui/pull/5738))
- Fixed `EuiDataGrid` cell focus sometimes not being restored for keyboard users when cell expansion popovers were closed ([#5761](https://github.com/elastic/eui/pull/5761))

**Breaking changes**

- Removed the `closePopover()` callback passed to `EuiDataGrid`'s `cellActions` render functions. Use `closeCellPopover()` passed by `EuiDataGrid`'s `ref` prop instead. ([#5734](https://github.com/elastic/eui/pull/5734))

**CSS-in-JS conversions**

- Converted `EuiAvatar` to CSS-in-JS styling ([#5670](https://github.com/elastic/eui/pull/5670))

cee-chen committed:
Upgrade EUI to v54.0.0

cee-chen committed:
[Discover] Remove deprecated closePopover call
- for closeCellPopover ref API

cee-chen committed:
[Lens] Remove deprecated closePopover call
- for closeCellPopover ref API

cee-chen committed:
[Security/Timelines] Remove deprecated closePopover call
- for closeCellPopover ref API

cee-chen committed:
[Security/Timeline] Update Timeline datagrid to accept/pass `visibleCellActions` prop

+ update Security to show 3 visible cell actions

cee-chen committed:
[APM] Account for removed EUI theme avatar sizes

constancecchen commented on x-pack/plugins/security_solution/public/common/components/events_viewer/index.tsx:
```diff
@@ -212,6 +212,7 @@ const StatefulEventsViewerComponent: React.FC<Props> = ({
               dataProviders,
               dataViewId,
               defaultCellActions,
+              visibleCellActions: 3,
```
@MikePaquette (pinging you because you posted in https://github.com/elastic/eui/pull/5675#issuecomment-1062902133)

I'm not overly familiar with Security's usage of EuiDataGrid - can you confirm that this component (Events Viewer?) is the correct/only component that needs 3 visible cell actions instead of the default 2?

chandlerprall replied:
From a slack thread:

> imo would be safer to just pass a hard coded 3 here https://github.com/elastic/kibana/blob/main/x-pack/plugins/timelines/public/components/t_grid/body/index.tsx#L819-#L839 instead of as a prop from the different places it’s used

andrew-goldstein replied:
> @MikePaquette (pinging you because you posted in [elastic/eui#5675 (comment)](https://github.com/elastic/eui/pull/5675#issuecomment-1062902133))
> 
> I'm not overly familiar with Security's usage of EuiDataGrid - can you confirm that this component (Events Viewer?) is the correct/only component that needs 3 visible cell actions instead of the default 2?

@constancecchen yes, this is the only component that needs 3 visible cell actions instead of the default 2

cee-chen committed:
Update emotion dependencies to latest

cee-chen committed:
Remove styles from being rendered in emotion serializer

cee-chen committed:
Update snapshot changes caused by EuiFormControlLayout changes

cee-chen committed:
Update snapshot changes caused by EuiAvatar CSS-in-JS conversion

thompsongl committed:
consolidate yarn.lock

cee-chen committed:
Update snapshots affected by emotion serializer `includeStyles: false` change

cee-chen committed:
[Spaces] Fix failing test due to new EuiAvatar emotion wrapper

- which, due to mount() causes .first() to no longer work as expected - targeting .last() instead gets the actual div element which works

cee-chen committed:
[Security] Fix cell expansion popover actions

- EUI added 2 `.euiPopoverFooter`s for overflowing cell actions, and Security's CSS to hide the first 2 cell actions (replaced by their own custom cell actions) was unintentionally affecting other actions

elasticmachine commented:
Pinging @elastic/eui-design (EUI)

sabarasaba reviewed (approved):
@elastic/platform-deployment-management changes lgtm

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

kqualters-elastic commented on x-pack/plugins/timelines/public/components/t_grid/body/index.tsx:
```diff
@@ -720,6 +728,7 @@ export const BodyComponent = React.memo<StatefulBodyProps>(
         columnHeaders,
         data,
         defaultCellActions,
+        visibleCellActions,
```
instead of passing this down as a prop from all the places it's used, wouldn't it be safer to just hard code the 3 in this component?

constancecchen replied:
Yeah that would be fine! I wasn't sure what all plugins used this TGrid component and if future usages might want to customize `visibleCellActions`, so was trying to play it safe. If you're confident all future usages will want a constant max of 3 visible cell actions I'm good with hard-coding it.

constancecchen replied:
https://github.com/elastic/kibana/pull/129653/commits/fc934748cd11ec83daef3dd10c6cd1ac97a65609 @kqualters-elastic @andrew-goldstein. Here's the relevant line where `3` is hard-coded:

https://github.com/elastic/kibana/blob/fc934748cd11ec83daef3dd10c6cd1ac97a65609/x-pack/plugins/timelines/public/components/t_grid/body/index.tsx#L719

Heenawter reviewed (approved):
Code review - primarily just adding new classes to Storybook components, plus some other minor restructuring of Color Manager Storybook. Presentation changes LGTM 👍

Commit:
Clean up spaces test snapshots

jportner commented on x-pack/plugins/spaces/public/management/spaces_grid/__snapshots__/spaces_grid_page.test.tsx.snap:
```diff
@@ -145,11 +145,90 @@ Array [
-          aria-hidden="true"
+        <Insertion
+          cache={
+            Object {
```
We spoke offline and I pushed the changes here: 3ddd8bde48f933f832994b5acb88b1f693bf38d8

thompsongl committed:
Merge branch 'main' into eui-54.0.0

andrew-goldstein reviewed (approved):
Thanks @constancecchen for upgrading the Security Solution to use `EuiDataGrid`'s new `visibleCellActions` option, such that there's no change in the behavior of the app 🙏 

- ✅  Desk tested locally
- ++ to @kqualters-elastic's [suggestion](https://github.com/elastic/kibana/pull/129653#discussion_r847438079) that configurability of the `2` -> `3` override via props isn't required
- While desk testing, I found what appears to be an unreleated issue with `EuiDataGrid` popovers owning focus in Safari: https://github.com/elastic/eui/issues/5778

LGTM

kertal reviewed (approved):
DataDiscover.team code owned code LGTM, tested in Cloud, works as expected, I'm exited about that
<img width="471" alt="10kb less when viewing page load panel" src="https://user-images.githubusercontent.com/463851/162934608-4e76d7e3-9563-4872-9b43-d225cd828481.png">
More than 10kb less in Discover's page load bundle 🥳  ... just wondering, how comes?

cee-chen committed:
Merge remote-tracking branch 'upstream/main' into eui-54.0.0

cee-chen committed:
[Security feedback] Revert 793d208 and hard-code visibleCellActions

constancecchen commented:
> More than 10kb less in Discover's page load bundle 🥳 ... just wondering, how comes?

Honestly I'm not sure 🙈 I'm not seeing anything in the diffs that would significantly account for that, so my best guess is maybe Emotion dependency upgrades?

kevinlog reviewed (approved):
I see that @andrew-goldstein and @kqualters-elastic have also reviewed with checked out code in Security solution.  Onboarding-Lifecycle just owns a snapshot update used for testing, which LGTM! (tests would fail otherwise).

mbondyra reviewed (approved):
LGTM 👌🏼

tsullivan reviewed (approved):
app services changes LGTM

jportner reviewed (approved):
🎉

constancecchen commented:
Thanks everyone!!

<img src="https://media.giphy.com/media/XgN7BVqzswKxrLWNkj/giphy-downsized.gif" alt="Picture of dog holding another dog back with caption 'Teamwork makes the dreamwork'" />
